### PR TITLE
Commands as a service

### LINKED ISSUE #189: Auto-registration of commands deprecated since Symfony 3.4
Example: `Auto-registration of the command "Symfony\Bundle\SwiftmailerBundle\Command\NewEmailCommand" is deprecated since Symfony 3.4 and won't be supported in 4.0. Use PSR-4 based service discovery instead`

### PR DESCRIPTION
**Pull request by ro0NL**
See https://github.com/symfony/symfony/issues/23488
Closes #189 

Tend to leave these container aware for now.

### CODE REVIEW

File: Resources/config/console.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <defaults public="false" />
```
Comment (stof) [inline Resources/config/console.xml]: I suggest avoiding to use the defaults here for now. The bundle currently supports version older than 3.3. Forcing us to maintain multiple versions in parallel just to use defaults for 3 private services is not worth it IMO.

**Comment by chalasr** [comment]
LGTM but we need to fix the low deps build... dunno why it fails for `3.4@dev` but not for `3.3@dev`

**Comment by chalasr** [comment]
@ro0NL could you add the static `$defaultName` to these commands? hope tests turn green magically

**Comment by ro0NL** [comment]
Lets see :)

**Comment by chalasr** [comment]
Seems like it did not... dunno :/

**Comment by chalasr** [comment]
@fabpot do you have any idea about what to do regarding the test suite? It can't checkout console 3.4 in `deps=low` as it conflicts with `symfony/symfony:2.8.*`.

File: DependencyInjection/SwiftmailerExtension.php [hunk 2]
```diff
@@ -48,6 +49,10 @@ public function load(array $configs, ContainerBuilder $container)
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('swiftmailer.xml');
 
+        if (class_exists(Application::class)) {
```
Comment (mvrhov) [inline DependencyInjection/SwiftmailerExtension.php]: This is available since sf 2.8. So this will get pulled in in all versions However the container build before 3.4 will fail because the commands there need to be public.

**Comment by ro0NL** [comment]
@chalasr looks like composer is passing :)

I believe this is compatible with 2.7 + 3.4@dev. Ill let @fabpot decide if we want to keep the lower bounds in composer ;-)

Failures related to upper bounds;

```
Fetching the "swiftmailer.mailer.default.transport" private alias is deprecated and will fail in Symfony 4.0. Make the alias public instead: 6x
```

File: Command/NewEmailCommand.php [hunk 3]
```diff
@@ -33,7 +35,7 @@ class NewEmailCommand extends ContainerAwareCommand
     protected function configure()
     {
         $this
-            ->setName('swiftmailer:email:send')
+            ->setName('swiftmailer:email:send') // BC with 2.7
```
Comment (fabpot) [inline Command/NewEmailCommand.php]: Can we use `self::$defaultName` here instead?

File: composer.json [hunk 4]
```diff
@@ -18,12 +18,12 @@
     "require": {
         "php": ">=7.0.0",
         "swiftmailer/swiftmailer": "^6.0.1",
-        "symfony/dependency-injection": "~2.7|~3.0",
+        "symfony/dependency-injection": "~2.7|~3.4@dev",
```
Comment (fabpot) [inline composer.json]: We should probably allow 4.0 deps for all deps as well, perhaps something to do in another PR before merging this one?

File: Command/NewEmailCommand.php (hunk 3 above)
Comment (ro0NL) [reply Command/NewEmailCommand.php]: nice catch. Yes we can :) doing it the lazy way right now.

File: composer.json (hunk 4 above)
Comment (ro0NL) [reply composer.json]: If that's OK, yes.

**Comment by ro0NL** [comment]
@fabpot i used `static::$defaultName` BTW. Due protected/non final state.

File: composer.json (hunk 4 above)
Comment (fabpot) [reply composer.json]: Yep, if you can create a PR and check that all tests still pass with 4.0, that would be great.

Comment (ro0NL) [reply composer.json]: lower bound is 3.x then right? or are we aiming for `2.7|3.4@dev|4.0@dev` something like that?

Comment (fabpot) [reply composer.json]: indeed, we should make it work for 2.7, 3.4+ and 4.0+

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Comment by ro0NL** [comment]
Ehm.. did you see latests commit?
